File: tests/fixtures/secrets/insecure_secrets.py
"""Fixture: Python file with multiple secret issues."""

import os

# SEC-001: Placeholder secrets
API_KEY = "your-api-key-here"
SECRET_KEY = "changeme"
JWT_SECRET = "supersecret"

# SEC-002: Low-entropy hardcoded secrets
DB_PASSWORD = "password123"
AUTH_TOKEN = "abc123def"

# SEC-003: Connection strings with embedded credentials
DATABASE_URL = "postgresql://admin:secretpass@db.example.com:5432/myapp"
REDIS_URL = "redis://user:redispass@redis.example.com:6379/0"

# SEC-004: Env var with default value
STRIPE_KEY = os.environ.get("STRIPE_SECRET_KEY", "sk_test_abc123xyz")
DB_PASS = os.getenv("DB_PASSWORD", "devpassword123")

# This should NOT trigger — proper env var usage
SAFE_KEY = os.environ["API_KEY"]
SAFE_SECRET = os.getenv("SECRET_KEY")

File: tests/fixtures/secrets/insecure_secrets.js
/**
 * Fixture: JavaScript file with secret issues.
 */

// SEC-001: Placeholder
const apiKey = "your-api-key-here";
const secretKey = "TODO_replace_this";

// SEC-002: Low-entropy secret
const password = "admin123";

// SEC-003: Connection string
const dbUrl = "mongodb://admin:mongopass123@mongo.example.com:27017/myapp";

// SEC-004: Env with default
const jwtSecret = process.env.JWT_SECRET || "devsecret123";
const dbPassword = process.env.DB_PASSWORD || "localpassword";

// This should NOT trigger — proper env var usage
const safeKey = process.env.API_KEY;
